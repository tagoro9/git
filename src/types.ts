export interface GitRemote {
  name: string;
  refs: {
    fetch: string;
    push: string;
  };
}

export enum GitErrorType {
  NOT_A_GIT_REPO,
  REMOTE_NOT_FOUND,
}
