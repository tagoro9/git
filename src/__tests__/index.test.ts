import simplegit = require("simple-git/promise");
import { git } from "~/index";
import { GitErrorType } from "~/types";

const simpleGitMock = (simplegit as unknown) as jest.Mock;

jest.mock("simple-git/promise");

const branchName = "f/my-fancy-branch";

const gitMocks = {
  revparse: jest.fn().mockResolvedValue(branchName),
};

describe("git", () => {
  beforeEach(() => {
    simpleGitMock.mockReturnValue({ silent: () => gitMocks });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("getCurrentBranchName", () => {
    test("returns the current branch name", async () => {
      await expect(git.getCurrentBranchName()).resolves.toMatchInlineSnapshot(
        `"f/my-fancy-branch"`
      );
      expect(gitMocks.revparse).toHaveBeenCalled();
      expect(gitMocks.revparse.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          Array [
            "--abbrev-ref",
            "HEAD",
          ],
        ]
      `);
    });

    test("throws an error when not on a git repo", async () => {
      gitMocks.revparse.mockRejectedValue(
        new Error(
          "fatal: not a git repository (or any of the parent directories): .git"
        )
      );
      await expect(git.getCurrentBranchName()).rejects.toHaveProperty(
        "code",
        GitErrorType.NOT_A_GIT_REPO
      );
    });
  });
});
