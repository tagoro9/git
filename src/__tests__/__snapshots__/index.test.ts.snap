// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git getRemote should fall back to the first remote if it cannot find the specified one 1`] = `
Object {
  "commit": undefined,
  "filepath": "",
  "filepathtype": "",
  "full_name": "tagoro9/git",
  "git_suffix": true,
  "hash": "",
  "href": "git@github.com:tagoro9/git.git",
  "name": "git",
  "organization": "tagoro9",
  "owner": "tagoro9",
  "pathname": "/tagoro9/git.git",
  "port": null,
  "protocol": "ssh",
  "protocols": Array [],
  "query": Object {},
  "ref": "",
  "resource": "github.com",
  "search": "",
  "source": "github.com",
  "toString": [Function],
  "token": "",
  "user": "git",
}
`;

exports[`git getRemote should return the parsed remote 1`] = `
Object {
  "commit": undefined,
  "filepath": "",
  "filepathtype": "",
  "full_name": "tagoro9/git",
  "git_suffix": true,
  "hash": "",
  "href": "git@github.com:tagoro9/git.git",
  "name": "git",
  "organization": "tagoro9",
  "owner": "tagoro9",
  "pathname": "/tagoro9/git.git",
  "port": null,
  "protocol": "ssh",
  "protocols": Array [],
  "query": Object {},
  "ref": "",
  "resource": "github.com",
  "search": "",
  "source": "github.com",
  "toString": [Function],
  "token": "",
  "user": "git",
}
`;

exports[`git getRemote should throw an error if there are no remotes 1`] = `"Could not find a remote with name \\"origin\\""`;
